src/main/java/org/apache/sysml/parser/ParameterizedBuiltinFunctionExpression.java

```diff
@@ -307,7 +314,65 @@ public void validateExpression(MultiAssignmentStatement stmt, HashMap<String, Da
+			raiseValidateError("Should provide more arguments for function " + fname, conditional, LanguageErrorCodes.INVALID_PARAMETERS);
+		}
+		//check for invalid parameters
+		Set<String> valid = UtilFunctions.asSet(Statement.PS_FEATURES, Statement.PS_LABELS, Statement.PS_VAL_FEATURES, Statement.PS_VAL_LABELS, Statement.PS_UPDATE_FUN, Statement.PS_AGGREGATION_FUN, Statement.PS_MODE, Statement.PS_UPDATE_TYPE, Statement.PS_FREQUENCY, Statement.PS_EPOCHS, Statement.PS_BATCH_SIZE, Statement.PS_PARALLELISM, Statement.PS_SCHEME, Statement.PS_HYPER_PARAMS, Statement.PS_CHECKPOINTING);
+		Set<String> params = new HashSet<>(getVarParams().keySet());
+		params.removeIf(Objects::isNull);
+		checkInvalidParameters(getOpCode(), params, valid);
+
+		// check existence and correctness of parameters
+		checkNonParameterizedDataType(fname, Statement.PS_MODEL, DataType.LIST, conditional); // check the model which is the only non-parameterized argument
+		checkDataType(fname, Statement.PS_FEATURES, DataType.MATRIX, conditional);
+		checkDataType(fname, Statement.PS_LABELS, DataType.MATRIX, conditional);
+		checkDataType(fname, Statement.PS_VAL_FEATURES, DataType.MATRIX, conditional);
+		checkDataType(fname, Statement.PS_VAL_LABELS, DataType.MATRIX, conditional);
+		checkDataValueType(false, fname, Statement.PS_UPDATE_FUN, DataType.SCALAR, ValueType.STRING, conditional);
+		checkDataValueType(false, fname, Statement.PS_AGGREGATION_FUN, DataType.SCALAR, ValueType.STRING, conditional);
+		checkStringParam(false, fname, Statement.PS_MODE, UtilFunctions.asSet(Statement.PS_MODE_LOCAL, Statement.PS_MODE_REMOTE_SPARK), conditional);
+		checkStringParam(false, fname, Statement.PS_UPDATE_TYPE, UtilFunctions.asSet(Statement.PS_UPDATE_TYPE_ASP, Statement.PS_UPDATE_TYPE_BSP, Statement.PS_UPDATE_TYPE_SSP), conditional);
+		checkStringParam(false, fname, Statement.PS_FREQUENCY, UtilFunctions.asSet(Statement.PS_FREQUENCY_BATCH, Statement.PS_FREQUENCY_EPOCH), conditional);
+		checkDataValueType(false, fname, Statement.PS_EPOCHS, DataType.SCALAR, ValueType.INT, conditional);
+		checkDataValueType(true, fname, Statement.PS_BATCH_SIZE, DataType.SCALAR, ValueType.INT, conditional);
+		checkDataValueType(false, fname, Statement.PS_PARALLELISM, DataType.SCALAR, ValueType.INT, conditional);
+		checkStringParam(false, fname, Statement.PS_SCHEME, UtilFunctions.asSet(Statement.PS_SCHEME_DC, Statement.PS_SCHEME_DRR, Statement.PS_SCHEME_DR, Statement.PS_SCHEME_OR), conditional);
+		checkDataValueType(true, fname, Statement.PS_HYPER_PARAMS, DataType.LIST, ValueType.UNKNOWN, conditional);
+		checkStringParam(true, fname, Statement.PS_CHECKPOINTING, UtilFunctions.asSet(Statement.PS_CHECKPOINTING_NONE, Statement.PS_CHECKPOINTING_EPOCH, Statement.PS_CHECKPOINTING_EPOCH10), conditional);
+
+		// set output characteristics
+		output.setDataType(DataType.LIST);
+		output.setValueType(ValueType.UNKNOWN);
+		output.setDimensions(getVarParam(null).getOutput().getDim1(), 1);
```

Comment:
Set the row dimension to the row dimension of the input parameter Statement.PS_MODEL if specified.

Reply:
In fact, I want to make a change in the API. I would like to ask the user to always put the parameterized "model=modelList" instead of the non-parameterized "modelList" as the first argument. I think it would be more simple and all the arguments are in the same format.